app/src/main/java/org/mozilla/focus/fragment/WebFragment.kt

```diff
@@ -49,17 +49,13 @@ abstract class WebFragment : LocaleAwareFragment() {
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         val view = inflateLayout(inflater, container, savedInstanceState)
 
-        webView = view.findViewById<View>(R.id.webview) as IWebView
-        isWebViewAvailable = true
-        webView!!.callback = createCallback()
-
-        val session = getSession()
-
-        if (session != null) {
-            webView!!.setBlockingEnabled(session.isBlockingEnabled)
+        webView = (view.findViewById<View>(R.id.webview) as IWebView).apply {
+            callback = createCallback()
+            setBlockingEnabled(session.isBlockingEnabled)
         }
+        isWebViewAvailable = true
 
-        if (session == null || !session.hasWebViewState()) {
+        if (!session.hasWebViewState()) {
```

Comment:
Okay, so we don't need this null check anymore? It shouldn't be a `!session?.hasWebViewState()` ? I'm curious why it was there in the first place then...

EDIT: Looked in the history, this was added to handle multiple tabs, so this should be fine.

Reply:
I think session should never be null but because we were using Java, it was hard to guarantee that and we had inconsistent code where session was sometimes checked and sometimes not. Now we can guarantee that. :)

I also think there was a separate BrowserFragment instance for each session, hence the final nature of it.